R/plot.r

```diff
@@ -36,43 +39,40 @@
 #'   evaluation.
 #' @export
 #' @examples
-#' # Generate some sample data, then compute mean and standard deviation
-#' # in each group
+#' # Create a data frame with some sample data, then create a data frame
+#' # containing the mean for each group in the sample data.
 #' set.seed(1)
-#' df <- data.frame(
-#'   gp = factor(rep(letters[1:3], each = 10)),
-#'   y = rnorm(30)
+#' sample_df <- data.frame(
+#'   group = factor(rep(letters[1:3], each = 10)),
+#'   value = rnorm(30)
 #' )
-#' ds <- do.call(rbind, lapply(split(df, df$gp), function(d) {
-#'   data.frame(mean = mean(d$y), sd = sd(d$y), gp = d$gp)
+#' group_means_df <- do.call(rbind, lapply(split(sample_df, sample_df$group), function(d) {
+#'   data.frame(group_mean = mean(d$value), group = d$group)
 #' }))
-#'
-#' # The summary data frame ds is used to plot larger red points on top
-#' # of the raw data. Note that we don't need to supply `data` or `mapping`
-#' # in each layer because the defaults from ggplot() are used.
-#' ggplot(df, aes(gp, y)) +
+#' 
+#' # Pattern 1
+#' # The group means data frame is used to plot larger red points on top
+#' # of the sample data. Note that we don't need to supply `data =` or `mapping =`
+#' # in each layer because the arguments are passed into ggplot() in the default
+#' # positions.
+#' ggplot(sample_df, aes(x = group, y = value)) +
 #'   geom_point() +
-#'   geom_point(data = ds, aes(y = mean), colour = 'red', size = 3)
-#'
+#'   geom_point(group_means_df, aes(y = group_mean), colour = 'red', size = 3)
+#' 
+#' # Pattern 2
 #' # Same plot as above, declaring only the data frame in ggplot().
 #' # Note how the x and y aesthetics must now be declared in
 #' # each geom_point() layer.
-#' ggplot(df) +
-#'   geom_point(aes(gp, y)) +
-#'   geom_point(data = ds, aes(gp, mean), colour = 'red', size = 3)
-#'
-#' # Alternatively we can fully specify the plot in each layer. This
-#' # is not useful here, but can be more clear when working with complex
-#' # mult-dataset graphics
+#' ggplot(sample_df) +
+#'   geom_point(aes(x = group, y = value)) +
+#'   geom_point(group_means_df, aes(x = group, y = group_mean), colour = 'red', size = 3)
+#' 
+#' # Pattern 3
+#' # Alternatively, we can fully specify the plot in each layer. This
+#' # can be particularly useful when working with complex, multi-dataset graphics.
 #' ggplot() +
-#'   geom_point(data = df, aes(gp, y)) +
-#'   geom_point(data = ds, aes(gp, mean), colour = 'red', size = 3) +
-#'   geom_errorbar(
-#'     data = ds,
-#'     aes(gp, mean, ymin = mean - sd, ymax = mean + sd),
-#'     colour = 'red',
-#'     width = 0.4
-#'   )
```

Comment:
It is not very clear to me why the errorbar was taken out of this example.

Reply:
It felt to me that the point of the examples was to show three different patterns for constructing the same graphic, and I thought the the error bars made that more confusing than illuminating. 

Would a comment above the three examples like this be helpful to further clarify the intent?

    # The following three code blocks create the same graphic, each using a different
    # pattern. In each graphic, the group means data frame is used to plot larger red
    # points on top of the sample data.
    
Or perhaps that wasn't the over-arching point. I can also now see that the error bars are meant to be an example of a "complex, multi-dataset graphic." But maybe showing the simpler versions of the three patterns would be easier for new learners?

Reply:
I don't think this comment is necessary because if people are executing the examples, it should be relatively straightforward to see how the plot relates to their input data. 

> Or perhaps that wasn't the over-arching point. I can also now see that the error bars are meant to be an example of a "complex, multi-dataset graphic." But maybe showing the simpler versions of the three patterns would be easier for new learners?

Yeah I like the clarity of the 3 patterns in these proposed changes and I agree that the patterns should be consistent. Perhaps a fourth example can be serviceable here to show how one could mix and match patterns, which may look like this:

```r
# When composing multi-dataset plots, it may be convenient to set the most
# frequently used dataset, along with shared aesthetics, in the main `ggplot()`
# function.
ggplot(group_means_df, aes(x = group)) +
 geom_point(aes(y = value), data = sample_df) +
 geom_point(aes(y = group_mean), colour = 'red', size = 3) +
 geom_errorbar(aes(ymin = group_mean - sd, ymax = group_mean + sd),
               colour = "red", width = 0.4)
```
(assuming that `sd` is also calculated)